executor/new_distsql.go

```diff
@@ -286,62 +289,196 @@ type IndexLookUpExecutor struct {
 	ctx       context.Context
 	schema    *expression.Schema
 	// This is the column that represent the handle, we can use handleCol.Index to know its position.
-	handleCol *expression.Column
-
-	// result returns one or more distsql.PartialResult.
-	result distsql.SelectResult
-
-	taskChan chan *lookupTableTask
-	tasksErr error
-	taskCurr *lookupTableTask
-
+	handleCol    *expression.Column
 	tableRequest *tipb.DAGRequest
 	// columns are only required by union scan.
 	columns  []*model.ColumnInfo
 	priority int
+	// All fields above is immutable.
+
+	indexHandler
+	tableHandler
 	finished chan struct{}
 }
 
+// indexHandler is used by IndexLookUpExecutor to maintain index lookup background goroutines.
+type indexHandler struct {
+	wg sync.WaitGroup
+}
+
+// openIndexHandler open the indexHandler, launch some background worker goroutines and write the results to workCh.
```

Comment:
It only launch one background goroutine to fetch handles.
send the results to workCh.